bynder_sdk/client/upload_client.py

```diff
@@ -1,144 +1,139 @@
-            while part_bytes:
-                part_nr = part_nr + 1
-                part_data = build_part_data(part_nr)
-
-                response = self.session.post(
-                    self.upload_url,
-                    files={'file': part_bytes},
-                    data=part_data['multipart_params'],
-                )
-                response.raise_for_status()
-                self._register_part(part_data, part_nr)
-                part_bytes = f.read(MAX_CHUNK_SIZE)
-
-        return part_data, total_parts
-
-    def _init_upload(self, filename, total_parts):
-        """ Gets the URL of the Amazon S3 bucket-endpoint in the region
-        closest to the server and initialises a file upload with Bynder
-        and returns authorisation information to allow uploading to the
-        Amazon S3 bucket-endpoint.
-        """
-        self.upload_url = self.session.get('/upload/endpoint/')
-
-        data = self.session.post(
-            '/upload/init/',
-            data={'filename': filename}
-        )
-        data['multipart_params'].update({
-            'chunks': total_parts,
-            'name': filename,
-        })
-
-        key = '{}/p{{}}'.format(data['s3_filename'])
-
-        def part_data(part_nr):
-            data['s3_filename'] = key.format(part_nr)
-            data['multipart_params'].update({
-                'chunk': part_nr,
-                'Filename': key.format(part_nr),
-                'key': key.format(part_nr),
-            })
-            return data
-
-        return part_data
-
-    def _register_part(self, init_data, part_nr):
-        """ Registers an uploaded chunk in Bynder.
+        response = self.session.post('/v7/file_cmds/upload/prepare')
+        return response["file_id"]
+
+    def _upload_chunks(self, file_path, file_id):
+        """Upload the file in chunks of MAX_CHUNK_SIZE.
+        :param file_path: The path of the asset to be uploaded.
+        :param file_id: The uuid4 used to identify the file to be uploaded.
+        :return:
+            - chunks_count: The number of chunks in which the file is to
+        be uploaded.
+            - file_size: The size of the file to be uploaded.
         """
+        chunks_count = 0
+        file_size = 0
+        with open(file_path, "rb") as f:
+            file_size = os.stat(f.fileno()).st_size
+            chunks_count = math.ceil(
+                file_size / MAX_CHUNK_SIZE)
+            chunk_nr = 0
+            for chunk in _read_in_chunks(f):
+                self._upload_chunk(file_id, chunk, chunk_nr)
+                chunk_nr = chunk_nr + 1
+
+        return chunks_count, file_size
+
+    def _upload_chunk(self, file_id, data, chunk_nr):
+        upload_chunk_endpoint = '/v7/file_cmds/upload/{}/chunk/{' \
+                                '}'.format(file_id, chunk_nr)
+        content_sha256 = {
+            'content-sha256': '{}'.format(sha256(data).hexdigest())
+        }
+        self.session.headers.update(content_sha256)
```

Comment:
```suggestion
        self.session.headers['content-sha256'] = sha256(data).hexdigest()
```
(assuming `hexdigest()` returns a string)